run_sim.py

```diff
@@ -0,0 +1,231 @@
+    # a Python string formatting pattern and format it in the loop.
+    # Note that click default `slurm_overrride` to (), and we want it to be "2D" tuple
+    slurm_override = slurm_override + (
+        ("job-name", "{sky_model}-fch{fch:04d}-chunk{ch}"),
+    )
+    # Make the SBATCH script minus hera-sim-vis.py command
+    program = _get_sbatch_program(hpc_config, slurm_override) if hpc_config else ""
+
+    # Option for hera-sim-vis.py. Let's just keep this fixed.
+    sim_options = f"--normalize_beams --fix_autos --log-level {log_level}"
+    sim_options += " --dry-run" if dry_run else ""
+
+    freq_chans = np.arange(*freq_range)
+    if freqs:
+        extra_freqs = freqs[np.isin(freqs, freq_chans, invert=True)]
+        freq_chans = np.append(freq_chans, extra_freqs)
+    logger.info(f"Frequency channels to run: {freq_chans}")
+
+    if force_remake_obsparams:
+        logger.info("Remaking all obsparams")
+        _make_h4c_obsparam(freq_range, freqs, sky_model, chunks)
+
+    for fch in freq_chans:
+        for ch in range(chunks):
+            logger.info(f"Working on frequency channel {fch} chunk {ch}")
+            # TODO: later remove nt17280
+            outfile = out_dir / f"{sky_model}_fch{fch:04d}_nt17280_chunk{ch}.uvh5"
+            # Check if output file already existed, if clobber is False
+            if skip_existing and outfile.exists():
+                logger.warning(f"File {outfile} exists, skipping")
+            else:
+                # TODO: later remove nt1780
+                obsparam = config_dir / f"fch{fch:04d}_chunk{ch}.yaml"
+                if not obsparam.exists():
+                    if make_missing_obsparam:
+                        _make_h4c_obsparam(
+                            freq_range=(fch, fch + 1),
+                            freqs=None,
+                            sky_model=sky_model,
+                            chunks=chunks,
+                        )
+                    else:
+                        logger.warning(f"{obsparam} does not exist, skipping")
+                else:
+                    cmd = f"hera-sim-vis.py {sim_options} {obsparam} {simulator_config}"
+                    if hpc_config:
+                        # Now, join the job script with the hera-sim-vis.py command
+                        # and format the job-name
+                        job_script = "\n".join([program, "", cmd, ""]).format(
+                            sky_model=sky_model, fch=fch, ch=ch
+                        )
+                        # Write job script and submit
+                        # Maybe we should save each job script individually
+                        with open("_sim.sbatch", "w") as fl:
+                            fl.write(job_script)
+
+                        subprocess.call(
+                            "sbatch _sim.sbatch".split()
+                        ) if not dry_run else None
+                        logger.info(f"\n===Job Script===\n{job_script}\n===END===\n")
```

Comment:
Yes, I would create a new directory, something like `batch_scripts/` and write each sbatch file into it, with the same name as the job name. That enables checking later.